src/stactools/core/utils/convert.py

```diff
@@ -16,25 +17,101 @@
+    :py:meth:`stactools.core.utils.convert.cogify_subdatasets`.
 
     Args:
         infile (str): The input file.
         outfile (str): The output COG to be written.
+        band (Optional[int]): The band number in the input file to extract.
+                            If not provided, a multi-band COG will be created.
         profile (Optional[dict[str, Any]]):
             An optional profile to use on the
             output file. If not provided,
             :py:const:`stactools.core.utils.convert.DEFAULT_PROFILE` will be
             used.
     """
-    if not utils.gdal_driver_is_enabled("COG"):
-        raise DriverRegistrationError(
-            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
-        )
-    destination_profile = DEFAULT_PROFILE.copy()
+    assert_cog_driver_is_enabled()
+
+    src = rasterio.open(infile)
+    dest_profile = DEFAULT_PROFILE.copy()
+    dest_profile.update(
+        {
+            "width": src.width,
+            "height": src.height,
+            "crs": src.crs,
+            "transform": src.transform,
+        }
+    )
+
     if profile:
-        destination_profile.update(profile)
-    rasterio.shutil.copy(infile, outfile, **destination_profile)
+        dest_profile.update(profile)
+
+    # If a band number was provided, create a single-band COG
+    if band:
+        single_band = src.read(band)
+        dest_profile.update({"count": 1, "dtype": single_band.dtype})
+        with rasterio.open(outfile, "w", **dest_profile) as dest:
+            dest.write(single_band, 1)
+    # If no band numbers were provided, create a multi-band COG
+    else:
+        dest_profile.update({"count": src.count, "dtype": src.dtypes[0]})
+        rasterio.shutil.copy(infile, outfile, **dest_profile)
+
+
+def cogify_subdatasets(
+    infile: str, outdir: str, subdataset_names: Optional[List[str]] = None
+) -> Tuple[List[str], List[str]]:
+    """Creates Cloud-Optimized GeoTIFFs for all subdatasets in a multi-dataset raster file.
+    The created files will be named the same as the source file, with a ``_SUBDATASET`` suffix.
+    E.g. if the source file is named ``foo.hdf`` and the subdataset is named ``bar``, the output
+    COG will be named ``foo_bar.tif``.
+     Args:
+         infile (str): The input file containing subdatasets.
+         outdir (str): The output directory where the COGs will be created.
+     Returns:
+         Tuple[List[str], List[str]]:
+             A two tuple (paths, names):
+                 - The first element is a list of the output COG paths
+                 - The second element is a list of subdataset names
+    """
+
+    with rasterio.open(infile) as dataset:
+        assert_cog_driver_is_enabled()
+        subdatasets = cast(List[str], dataset.subdatasets)
+        base_file_name = os.path.splitext(os.path.basename(infile))[0]
+        paths = []
+        names = []
+        for subdataset in subdatasets:
+            subd = rasterio.open(subdataset)
```

Comment:
I'm not sure if we need to re-open the subdataset with rasterio. However, I can't check, because there's no unit test for the subdataset case.

Reply:
I believe you do need to reopen the subdataset, since `dataset.subdatasets` only returns a list of strings containing the subdataset names. To loop through subdatasets and export each out as a separate file, you need to open each subdataset. I added a unit test for this function. 
Again, happy to be informed otherwise.